reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java

```diff
@@ -740,19 +676,18 @@ else if (close) {
 	@Test
 	void testIssue186() {
 		disposableServer =
-				HttpServer.create()
-				          .port(0)
+				createServer()
 				          .handle((req, res) -> res.status(200).send())
-				          .wiretap(true)
 				          .bindNow();
 
-		HttpClient client =
-				HttpClient.create(ConnectionProvider.create("testIssue186", 1))
-				          .remoteAddress(disposableServer::address)
-				          .wiretap(true);
+		ConnectionProvider provider = ConnectionProvider.create("testIssue186", 1);
+		HttpClient client = createClient(provider, disposableServer::address);
 
 		doTestIssue186(client);
 		doTestIssue186(client);
+
+		provider.disposeLater()
```

Comment:
in case of exception / assertion error, the `provider` wouldn't be disposed.

Reply:
fixed